wpilibj/src/test/java/edu/wpi/first/wpilibj/MockHardwareExtension.java

```diff
@@ -11,21 +11,15 @@
 
 import com.google.common.base.Stopwatch;
 
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
 import edu.wpi.first.wpilibj.hal.HAL;
 import edu.wpi.first.wpilibj.util.BaseSystemNotInitializedException;
 
-/**
- * Utility class for configuring unit tests.
- */
-public final class UnitTestUtility {
-  private UnitTestUtility() {
-        /* no-op */
-  }
-
-  /**
-   * Sets up the base system WPILib so that it does not rely on hardware.
-   */
-  public static void setupMockBase() {
+public final class MockHardwareExtension implements BeforeAllCallback {
+  @Override
+  public void beforeAll(ExtensionContext context) throws Exception {
     HAL.initialize(500, 0);
```

Comment:
@ThadHouse Should this be called before _every_ test?

Reply:
It needs to be called before any test runs. It doesn't have to be called before every one, but at least before the first test runs.